dev-support/Jenkinsfile

```diff
@@ -56,6 +56,34 @@ pipeline {
             }
         }
 
+        // This is an optional stage which runs only when there's a change in
+        // C++/C++ build/platform.
+        // This stage serves as a means of cross platform validation, which is
+        // really needed to ensure that any C++ related/platform change doesn't
+        // break the Hadoop build on Centos 8.
+        stage ('precommit-run Centos 8') {
+            environment {
+                DOCKERFILE = "${SOURCEDIR}/dev-support/docker/Dockerfile_centos_8"
+                IS_OPTIONAL = 1
+            }
+
+            steps {
+                withCredentials(
+                    [usernamePassword(credentialsId: 'apache-hadoop-at-github.com',
```

Comment:
This is now repeating 3 times in this file, can we reuse?

Reply:
@goiri I tried the following -
1. Putting the stages for Centos 8, Debian 10, Ubuntu Focal under a common credentials block - https://github.com/apache/hadoop/pull/3145/commits/b1cf25ce9d756c8a3c7cb5ec3b96555537c16ec2. This results in a syntax error.
2. Putting all the stages under the same credentials block - https://github.com/apache/hadoop/pull/3145/commits/f3b8b72b4c2af3f1584a7228d123049da8290f3c. Syntax error.
3. Creating credential environment variables that are scoped to the stages - https://github.com/apache/hadoop/pull/3145/commits/74acdc93c7cb6a5df612a81326f577e87f597c8b. This syntax isn't supported as well.
4. Moved these environment variables to the scope of the whole pipeline - https://github.com/apache/hadoop/pull/3145/commits/83d7399b75551a0b9e73a7ba116450a2964eb81e. This is not at all recommended since leads to poor management of credentials. Moreover, this didn't work.

I think there's no way to work around this one. Could we please proceed with the current approach for now?